### DISCUSSION TITLE
[REQUEST] Tree Inspect View

### DISCUSSION THREAD

**Discussion by rickdonato** [Textualize/rich#1962] [Ideas]
Consider posting in https://github.com/willmcgugan/rich/discussions for feedback before raising a feature request.

Have you checked the issues for a similar suggestions? Yes

**How would you improve Rich?**

Create a tree option for inspect. The reason for this request is this would be a great learning tool when dealing with nested structures. 
Each object in the tree could display the object type and name, along with the methods and attributes. An option could be provided to allow for keys to be followed, or just the objects. Or even this could be a UI tool to allow for the object to be explored. Much like `jless`.

**What problem does it solve for you?**
Helping learn about nested structures/obejcts in python without manually walking the path.

What problem do you have that this feature would solve? I may be able to suggest an existing way of solving it.

**Did I help**

If I was able to resolve your problem, consider [sponsoring](https://github.com/sponsors/willmcgugan) my work on Rich, or [buy me a coffee](https://ko-fi.com/willmcgugan) to say thanks.

**Comment by willmcgugan** [comment]
Might be useful, but not sure it belongs in the core lib. For the dynamic option, you could build that with Textual.

**Reply by rickdonato** [reply]
I can see both sides of not having it in the core lib. The only thing is. From what I can see Im not sure if I can get enough information directly out of the core _inspect to be able to build a separate tool. As most of the internal methods/work you are performing within _inspect are for the rendering. Would be great if the info from inspect could be exposed via Python i.e `dict()` etc so that the info could be pulled into additional tools folks may want to write.
